clients/python/girder_client/__init__.py

```diff
@@ -520,48 +574,43 @@ def uploadFile(self, parentId, stream, name, size, parentType='item',
         if reference is not None:
             params['reference'] = reference
         obj = self.post('file', params)
-        if '_id' in obj:
-            uploadId = obj['_id']
-        else:
+        if '_id' not in obj:
             raise Exception(
                 'After creating an upload token for a new file, expected '
                 'an object with an id. Got instead: ' + json.dumps(obj))
 
-        offset = 0
-        while True:
-            data = stream.read(self.MAX_CHUNK_SIZE)
-
-            if not data:
-                break
-
-            params = {
-                'offset': offset,
-                'uploadId': uploadId
-            }
-            files = {
-                'chunk': data
-            }
-            obj = self.post('file/chunk', parameters=params, files=files)
-            offset += len(data)
+        return self._uploadContents(
+            obj, stream, size, progressCallback=progressCallback)
 
-            if '_id' not in obj:
-                raise Exception('After uploading a file chunk, did'
-                                ' not receive object with _id. Got instead: ' +
-                                json.dumps(obj))
+    def uploadFileContents(self, file_id, stream, size, reference=None):
+        """
+        Uploads the contents of an existing file..
 
-            if callable(progressCallback):
-                progressCallback({
-                    'current': offset,
-                    'total': size
-                })
+        :param file_idd: ID of file to update
+        :param stream: Readable stream object.
+        :type stream: file-like
+        :param size: The length of the file. This must be exactly equal to the
+            total number of bytes that will be read from ``stream``, otherwise
+            the upload will fail.
+        :type size: str
+        :param reference: optional reference to send along with the upload.
+        :type reference: str
+        """
+        path = 'file/' + file_id + '/contents'
```

Comment:
Minor nit, but let's use `%s` interpolation for this var since it's in the middle of the string